dmtools-core/src/integrationTest/java/com/github/istin/dmtools/ai/agent/TestCaseDeduplicationAgentTest.java

```diff
@@ -0,0 +1,203 @@
+package com.github.istin.dmtools.ai.agent;
+
+import com.github.istin.dmtools.common.model.ToText;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class TestCaseDeduplicationAgentTest {
+
+    TestCaseDeduplicationAgent agent;
+
+    @Before
+    public void setUp() throws Exception {
+        agent = new TestCaseDeduplicationAgent();
+    }
+
+    @Test
+    public void testDeduplicationWithExactMatches() throws Exception {
+        try {
+            List<TestCaseGeneratorAgent.TestCase> newTestCases = new ArrayList<>();
+            newTestCases.add(new TestCaseGeneratorAgent.TestCase(
+                    "High",
+                    "User can login",
+                    "Test user login functionality"
+            ));
+            newTestCases.add(new TestCaseGeneratorAgent.TestCase(
+                    "Medium",
+                    "User can logout",
+                    "Test user logout functionality"
+            ));
+
+            String newTestCasesText = ToText.Utils.toText(newTestCases);
+            String existingTestCases = "Priority: High\nSummary: User can login\nDescription: Existing login test\n";
+
+            List<TestCaseGeneratorAgent.TestCase> result = agent.run(new TestCaseDeduplicationAgent.Params(
+                    newTestCasesText,
+                    existingTestCases,
+                    ""
+            ));
+
+            // Should only return the logout test case, as login already exists
```

Comment:
medium: The exception handling logic to skip tests is duplicated across multiple test methods. To improve maintainability and reduce code duplication, you could extract this logic into a private helper method.
```suggestion
    private void handleTestException(Exception e) throws Exception {
        if (e.getMessage() != null && (e.getMessage().contains("API") ||
                e.getMessage().contains("key") ||
                (e.getMessage().contains("response") && e.getMessage().contains("null")))) {
            System.out.println("Skipping test due to configuration issue: " + e.getMessage());
            return;
        }
        throw e;
    }
```
You can then call `handleTestException(e)` in your catch blocks.